thunder/tests/opinfos.py

```diff
@@ -3609,6 +3609,47 @@ def reshape_sample_generator(op, device, dtype, requires_grad, **kwargs):
 shape_ops.append(reshape_opinfo)
 
 
+def unflatten_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    cases = (
+        ((4,), 0, (4,)),  # no-op
+        ((2, 2), -1, (2,)),  # no-op
+        ((1, 2, 1), 1, (2,)),  # no-op
+        ((4, 2), 0, (2, 2)),
+        ((125, 3), 0, (25, 5)),
+        ((25, 25), -1, (1, 5, 5)),
+        ((16, 32), -1, (2, 4, 1, 4)),
+        ((4, 5, 6), -1, (6, 1, 1, 1)),
```

Comment:
PyTorch allows `-1` sizes. Let's test these as well?

Reply:
Adding tests for partial shape specification is great. Let's also add some tests of unflattening dimensions which are not the outermost or innermost, like `(4, 12, 6), 1, ...`

Reply:
Sure thing! I wasn't aware that `-1` sizes were allowed. Thanks so much!